app/Adapters/Task.php

```diff
@@ -28,6 +28,42 @@ public function process()
         $this->task->estimatedHours = sprintf('%.2f', $this->task->estimatedHours);
         $this->task->xp = sprintf('%.2f', $this->task->xp);
 
+        $taskStatus = $profilePerformance->perTask($this->task);
+
+        if (!empty($taskStatus)) {
+            $taskEstimatedSeconds = $mappedValues['estimatedHours'] * 60 * 60;
+            $taskDeliveredOnTime = $taskStatus[$this->task->owner]['workSeconds']
+            <= $taskEstimatedSeconds;
+
+            // deadline in last 25% of the time of task
+            $lastQuarterOfTask = 0.25 * $taskEstimatedSeconds;
+
+            //generate task color status
+            if ($this->task->passed_qa === true && $taskDeliveredOnTime === true) {
```

Comment:
This has to be last check either way since if for example task was wrapped up (qa passed) within last 25% of estimate, it task will be orange instead of green